### DISCUSSION TITLE
Show products based on customer group

### DISCUSSION THREAD

**Discussion by bakerhaider** [swellstores/community#438] [Help] [answered]
Hello im makingy own storefront which will have different groups and i want to show different products based on the users group, what is the best way to achieve this functionality?

**Comment by sags95** [comment, marked as answer]
Hi @BakerAlblam, you can likely do this a few ways. Using only the frontend API you can do something like the following

- On products, add a "customer group" attribute with a value for the customer group that the product should be linked to
- On the storefront, authenticate the customer, this will also return the `group` field that denotes the customer group.
- Fetch products with the `customer group` attribute value that matches the `group` id. It can be something like this:

```
await swell.products.list({
   $filters: {
      customer_group: '<customer_group>'
   }
});
```

**Reply by bakerhaider** [reply]
> Hi @BakerAlblam, you can likely do this a few ways. Using only the frontend API you can do something like the following
> 
> 
> 
> - On products, add a "customer group" attribute with a value for the customer group that the product should be linked to
> 
> - On the storefront, authenticate the customer, this will also return the `group` field that denotes the customer group.
> 
> - Fetch products with the `customer group` attribute value that matches the `group` id. It can be something like this:
> 
> 
> 
> ```
> 
> await swell.products.list({
> 
>    $filters: {
> 
>       customer_group: '<customer_group>'
> 
>    }
> 
> });
> 
> ```
> 
> 
> 
> 
> 
> 
> 
> 

Thanks, I'll look into it!
